### DISCUSSION TITLE
DFU - no clear moment to perform ECU reset

### DISCUSSION THREAD

**Discussion by gmikitiuk** [hathach/tinyusb#2878] [General]
Hello,

I'm currently working with DFU and I cannot clearly understand or find a point where ECU can perform reset after flashing has been done(all messages sent to the HOST are out). 

```
DFU state(7) = dfuMANIFEST, status(0) = No error condition is present
DFU state(8) = dfuMANIFEST-WAIT-RESET, status(0) = No error condition is present
Resetting USB to switch back to runtime mode
```

Is there any notification in tinyUSB which I can hook up to monitor "Resetting USB to switch back to runtime mode"?

I would appreciate some hints

**Comment by HiFiPhile** [comment]
Hi, if you use `dfu-util` you can set set a 5ms timer to ensure ack is received by PC then reboot in `tud_dfu_detach_cb()`

**Reply by gmikitiuk** [reply]
Hi,

I believe I was missing `-R` parameter when calling dfu-util, now I got proper detach callback.
